.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+repos:
+    -   repo: https://github.com/pre-commit/pre-commit-hooks
+        rev: v4.3.0  # Use the ref you want to point at
+        hooks:
+        -   id: trailing-whitespace
+        -   id: end-of-file-fixer
+        -   id: check-docstring-first
+    -   repo: https://github.com/pycqa/isort
+        rev: 5.10.1
+        hooks:
+        - id: isort
+          name: isort (python)
+    -   repo: https://github.com/pre-commit/pygrep-hooks
+        rev: v1.9.0  # Use the ref you want to point at
+        hooks:
+        -   id: python-use-type-annotations
+# This hook is still too restrictive. We should if we want to add it or not.
+#    -   repo: https://github.com/pycqa/pydocstyle
+#        rev: 4.0.0  # pick a git hash / tag to point to
+#        hooks:
+#        -   id: pydocstyle
+#
+#            -   --select=
+#                D100,
+#                D101,
+#                D102,
+#                D103,
+#                D104,
+#                D105,
+#                D106,
+#                D107,
```

Comment:
These are going to require us to do some code modification to get them all to pass, but I like them. I'd also like to see D200, D201, D204, D209, D210, D211, **D300**, D402, and D403 added here. In fact, the pep257 convention would be ideal, but it will report a lot of issues, so I left some of the more glaring ones out., such as having a blank line after a docstring, which I think is fine.

Reply:
I'll add a task to implement this.